### PR TITLE
BigQuery copy materialization enhancement

### LINKED ISSUE #3570: BigQuery 'copy' materialization doesn't work
Trying to check model:
```sql
{{ config(
    materialized='copy'
)}}

select * from {{ source('int', 'Order_delta') }}
```
 and receive following error:
 ```
Running with dbt=0.20.0
Found 2 models, 0 tests, 0 snapshots, 0 analyses, 166 macros, 0 operations, 0 seed files, 7 sources, 0 exposures

14:22:42 | Concurrency: 1 threads (target='poc')
14:22:42 | 
14:22:42 | 1 of 1 START copy model poc_dbt.OrderDedupe.......................... [RUN]
14:22:42 | 1 of 1 ERROR creating copy model poc_dbt.OrderDedupe................. [ERROR in 0.03s]
14:22:42 | 
14:22:42 | Finished running 1 copy model in 2.79s.

Completed with 1 error and 0 warnings:

Compilation Error in model OrderDedupe (models/int/OrderDedupe.sql)
  Copy table materialization must be "copy" or "table", but config.get('copy_materialization', 'table') was None

  > in macro materialization_copy_bigquery (macros/materializations/copy.sql)
  > called by model OrderDedupe (models/int/OrderDedupe.sql)

Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
```

There is no documentation for this materialization but it exists in the code.
Is there any way to get it work?

### PR DESCRIPTION
**Pull request by xemuliam**
resolves #3570

### Description

1. Add implicit usage of `copy_materialization='table'` if it is not found in config (simplification for end-user)
2. Align `adapter.table_copy` implementation with [BigQuery documentation](https://cloud.google.com/bigquery/docs/managing-tables#copying_multiple_source_tables) (ability to use several source tables to be copied into single target table at a time)
3. Change integration test to check (1) and to test (2) for two source tables (reference by `ref` and `source`)

### Checklist
 - [ X ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [ X ] I have run this code in development and it appears to resolve the stated issue (custom Docker image has been built from changed code and process of table(s) copying has been checked several times for different scenarios)
 - [ X ] This PR includes tests, or tests are not required/relevant for this PR
 - [ X ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

**Comment by xemuliam** [comment]
OK to test

**Comment by jtcohen6** [comment]
Thanks for the PR @xemuliam! I haven't had a chance to take this for a spin just yet, it's very much on my list.

File: plugins/bigquery/dbt/adapters/bigquery/connections.py [hunk 1]
```diff
@@ -458,26 +458,32 @@ def copy_bq_table(self, source, destination, write_disposition):
...
         client = conn.handle
 
-        source_ref = self.table_ref(
-            source.database, source.schema, source.table, conn)
+        if type(source) is not list:
+            source = [source]
+
+        source_ref_array = [self.table_ref(
+            src_table.database, src_table.schema, src_table.table, conn)
+            for src_table in source]
```
Comment (jtcohen6) [inline plugins/bigquery/dbt/adapters/bigquery/connections.py:466]: Appreciate the backwards compatibility here! Could you add a comment to the effect of, `source` used to expect only one table reference, but now we accept multiple?

File: test/integration/022_bigquery_test/copy-models/schema.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+version: 2
+sources:
+  - name: test_copy_several_tables
+    schema: "{{ target.schema) }}"
```
Comment (jtcohen6) [inline test/integration/022_bigquery_test/copy-models/schema.yml:4]:
```suggestion
    schema: "{{ target.schema }}"
```

Comment (jtcohen6) [reply test/integration/022_bigquery_test/copy-models/schema.yml:4]: After I fix this locally, and run `test_bigquery_copy_models.py`, I see `copy_as_several_tables` failing with:
```
Database Error in model copy_as_several_tables (copy-models/copy_as_several_tables.sql)
  Table dbt-test-env:test16303148312675888629_bigquery_test_022.original appears multiple times in source tables.
```
Do you have a sense of what that error indicates?

**Review by jtcohen6 — COMMENTED** [review]
@leahwicz @kwigley It looks like this PR skipped running all adapter integration tests (https://github.com/dbt-labs/dbt/actions/runs/1179035082), when it should have run BigQuery tests. Do we need to adjust a PR trigger?

@xemuliam Thanks for all the work here! It looks like there's still an error in the integration test. I'm not sure why it's not running in CI, but we'll definitely want to resolve that before proceeding.

**Comment by jtcohen6** [comment]
The relevant test (`022_bigquery_test/test_bigquery_copy_models.py`) is now passing for me locally. I'd still like to figure out why it's not running as part of automated CI.

File: test/integration/022_bigquery_test/copy-models/schema.yml (hunk 2 above)
Comment (xemuliam) [reply test/integration/022_bigquery_test/copy-models/schema.yml]: Fixed

File: plugins/bigquery/dbt/adapters/bigquery/connections.py (hunk 1 above)
Comment (xemuliam) [reply plugins/bigquery/dbt/adapters/bigquery/connections.py:474]: Added

**Review by jtcohen6 — APPROVED** [review]
@xemuliam Thank you so much for the contribution, and for all the iterative work to get it over the finish line! This change will be included in v0.21 :)

The next step is to actually, officially document the `copy` materialization (and probably the `copy_bq_table` method, too). Any chance you'd be up to contribute there, too? https://github.com/dbt-labs/docs.getdbt.com

**Pull request merged**